modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/plugin/json/GenericJsonDescriptorProperty.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal.plugin.json;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+/**
+ * Generic descriptor that will be used to describe parametrization to construct {@link ExtensionModel},
+ * {@link ClassLoader} and any other descriptor that may arise in a future of {@link MulePluginJsonDescriber}.
+ * <p/>
+ * Each {@link GenericJsonDescriptorProperty} will have an ID that will be used to discover any loader that's responsible of working
+ * with the current set of attributes. It's up to each loader to validate the types, size and all that matters around
+ * the attributes.
+ *
+ * @since 4.0
+ */
+public class GenericJsonDescriptorProperty {
```

Comment:
Not sure what should I put, this class will hold data from different sources.
Taking the following JSON snippet:
```
{
  "name": "plugin-name",
  "minMuleVersion": "4.2.3",
  "classLoaderDescriptor": {
    "id": "classloader",
    "attributes": {
      "exportedPackages": ["org.mule.extension.api"],
      "exportedResources": ["META-INF", "META-INF/otherstuff"]
    }
  },
  "extensionModelDescriptor": {
    "id": "annotations",
    "attributes": {
      "class": "org.mule.extension.internal.ExtensionConnector"
    }
  }
}
```
One instance will have ID->"classloader" with 2 attributes (exportedPackages and exportedResources) and the other instance will have an ID -> "annotations" with 1 attribute (the class).

Which will be held in the `MulePluginJsonDescriber` object (right now, it's just implemented the one for the `ExtensionModel`)